Doc implement reredirects (#19456)
matplotlib/matplotlib: doc/devel/documenting_mpl.rst

Miscellaneous
=============

Moving documentation
--------------------

Sometimes it is desirable to move or consolidate documentation.  With no
action this will lead to links either going dead (404) or pointing to old

> versions of the documentation.  Preferable is to make a redirect on the
> new page that refers to the old page.  This will make the old page, but it
> will have an html refresh that immediately redirects the viewer to the new
> page.

Reviewer: ```suggestion
versions of the documentation.  It is preferable is to make a redirect on the
old page that refers to the new page, which will have an html refresh that
immediately redirects the viewer to the new page.
```

Author: The redirect is on the page that still exists, the "new page" - the old rat is removed.

Reviewer: The redirect _directive_ is on the new page, but the redirect itself is on the (generated) old page. You can swap which one you want to refer to in this sentence from what I suggested, but at the moment, it is incorrect.

Author: Changed this paragraph to:

> Sometimes it is desirable to move or consolidate documentation.  With no
action this will lead to links either going dead (404) or pointing to old
versions of the documentation.  Preferable is to replace the old page
with an html refresh that immediately redirects the viewer to the new
page. So, for example we move ``/doc/topic/old_info.rst`` to
``/doc/topic/new_info.rst``.  We remove ``/doc/topic/old_info.rst`` and
in ``/doc/topic/new_info.rst`` we insert a ``redirect-from`` directive that
tells sphinx to still make the old file with the html refresh/redirect in it
(probably near the top of the file to make it noticeable)

So, for example we move ``/doc/topic/old_info.rst`` to

> ``/doc/topic/new_info.rst``.  We make remove ``/doc/topic/old_info.rst`` and

Reviewer 2: typo: "make remove"

in ``/doc/topic/new_info.rst`` we insert a redirect (probably near the
top of the file to make it noticeable)

.. code-block:: rst

   .. redirect-from:: /topic/old_info

In the built docs this will yield an html file
``/build/html/topic/old_info.html`` that has a refresh to ``new_info.html``.
If the two files are in different subdirectories:

.. code-block:: rst

   .. redirect-from:: /old_topic/old_info2

will yield an html file ``/build/html/old_topic/old_info2.html`` that has a
(relative) refresh to ``../topic/new_info.html``.

> Use the  full path for this directive, relative to the doc root at

Reviewer 2: one space too many before "full"

``http://matplotlib.org/stable/``.  So ``/old_topic/old_info2`` would be
found by users at ``http://matplotlib.org/stable/old_topic/old_info2``.

> Do not use relative links.

Reviewer 2: the intent was for relative links to work too, although it's admittedly untested...

Author: Yeah, that would likely be OK as well, but I think its a lot less confusing if we stick to full paths...
